### ISSUE TITLE
Validate typings using actual `tsc` compiler

### ISSUE THREAD

**Issue by robertrossmann** [strvcom/atlas.js#70]
The typings behave somewhat different when working on them inside this project and when they are being consumed as a module installed from `npm` (see #69).

We should add some kind of minimum TypeScript project which will try to consume the packages as if they were installed from npm and attempt to compile the TypeScript project to make sure the type definition files are actually working from a user perspective, not just from Atlas-maintainer perspective. 💪

**Comment by yunicz** [comment]
Yeah, this basically prevents us from using any typing when working with atlas, luckily you can simply overcome this issue and get your project compiled by requiring atlas instead of importing it
const { Atlas } = require('@atlas.js/atlas');

**Comment by robertrossmann** [comment]
@yunicz what is the problem when trying to import `@atlas.js/atlas`? I tried on at least two projects and the typings are available when I use import statements:

```js
import { Atlas } from '@atlas.js/atlas'
```

If you could share any TypeScript compilation errors that would be very helpful. 🙏

**Comment by yunicz** [comment]
nvm I am stupid, maybe just make componentOptions optional so that we don't have to write ` { aliases: {} }` everytime component has no deps
